engine/crates/engine-v2/src/operation/logical_planner/selection_set.rs

```diff
@@ -553,10 +606,153 @@ impl<'schema, 'a> SelectionSetLogicalPlanner<'schema, 'a> {
                     }
                 }
             }
+
+            // We did not find any candicates from unplanned fields. If our field we try to plan with
+            // is in an interface, which does not have keys set, we need to re-plan these fields from all
+            // the types implementing the interface.
+            match definition.parent_entity() {
+                schema::EntityDefinition::Interface(interface) if candidates.is_empty() => {
+                    let mut add_to_plan = Vec::new();
+
+                    for r#type in interface.possible_types() {
+                        if let Some(field) = r#type.fields().find(|field| field.name() == definition.name()) {
+                            add_to_plan.push(field);
+                        }
+                    }
+
+                    plan_changes.push((id, add_to_plan));
+                }
+                _ => (),
+            }
+        }
+
+        match planned_selection_set.id {
+            Some(selection_set_id) if !plan_changes.is_empty() => {
+                self.replan_fields(selection_set_id, plan_changes, unplanned_fields);
+                self.generate_all_candidates(unplanned_fields, planned_selection_set, candidates)?
+            }
+            _ => (),
         }
+
         Ok(())
     }
 
+    /// Replans the fields within a selection set by resetting their current plans and adding
+    /// new fields to be planned. This method may be invoked when the structure of the selection
+    /// set changes or when new fields become available for planning.
+    fn replan_fields(
+        &mut self,
+        selection_set_id: SelectionSetId,
+        plan_changes: Vec<(FieldId, Vec<FieldDefinition<'schema>>)>,
+        unplanned_fields: &mut HashMap<FieldId, FieldDefinition<'schema>>,
+    ) {
+        for (field_id, add_to_plan) in plan_changes {
```

Comment:
would have used `interface_field_id` instead of `field_id` and renamed the function `replan_interface_fields()`